<launch>
  <include file="$(find vicon_bridge)/launch/vicon.launch" />

  <node pkg="asctec_ll_updated" name="hummingbird_1" type="asctec_ll_updated_node" output="screen">
     <param name="port" value="/dev/ttyUSB0"/>
     <param name="ll_status_rate" value="5"/>
     <param name="imu_calc_data_rate" value="0"/>
     <param name="rc_data_rate" value="0"/>
     <param name="ctrl_out_rate" value="10"/>
     <param name="gps_data_advanced_rate" value="0"/>
  </node>

  <node name="joy_controller" pkg="joy" type="joy_node" output="screen"/>

  <node name="asctec_controller" 
        pkg="pc_asctec_sim" 
        type="controller_node">

     <rosparam file="$(find pc_asctec_sim)/cfg/default_h.yaml" command="load"/>
     <param name="name" value="/hummingbird_1"/>
     <param name="world_frame" value="/odom"/>
     <param name="quad_frame" value="/vicon/hummingbird_1/hummingbird_1"/>

  </node>

  <node name="ugv_tracker" 
        pkg="pc_asctec_sim" 
        type="ugv_tracker" 
        output="screen">

     <param name="quad_name" value="/hummingbird_1"/>
     <param name="quad_frame" value="/vicon/hummingbird_1/hummingbird_1"/>
     <param name="world_frame" value="/odom"/>
     <param name="ugv_frame" value="/vicon/jackal2/jackal2"/>
  </node>

  <node name="waypoint_server" 
        pkg="pc_asctec_sim" 
        type="trajectory_real_node">

     <param name="name" value="/hummingbird_1"/>
     <param name="world" value="/odom"/>
  </node>

</launch>
